toolchain/check/merge.cpp

```diff
@@ -450,12 +450,23 @@ static auto CheckRedeclParamSyntax(Context& context,
   for (; new_iter != new_end && prev_iter != prev_end;
        ++new_iter, ++prev_iter) {
     auto new_node_id = *new_iter;
+    auto new_node_kind = context.parse_tree().node_kind(new_node_id);
+    // Skip over "unused" markers.
+    if (new_node_kind == Parse::NodeKind::UnusedPattern) {
+      ++new_iter;
+      new_node_id = *new_iter;
+      new_node_kind = context.parse_tree().node_kind(new_node_id);
+    }
     auto prev_node_id = *prev_iter;
+    auto prev_node_kind = context.parse_tree().node_kind(prev_node_id);
+    if (prev_node_kind == Parse::NodeKind::UnusedPattern) {
+      ++prev_iter;
+      prev_node_id = *prev_iter;
+      prev_node_kind = context.parse_tree().node_kind(prev_node_id);
+    }
```

Comment:
Where is this logic tested? I would expect:

```
fn F(x: i32);
// Does not diagnose per https://github.com/carbon-language/carbon-lang/pull/2022/files#diff-4b37f3cfff497529f753862b14524eaf426840693ccddb49973eda06b0459958R133
fn F(unused x: i32) { ... }
```

However, and I think the logic is incorrect for this:

```
fn F(unused x: i32);
// Should diagnose, looks like it won't.
fn F(x: i32) { ... }
```